dune/grid/common/p2pcommunicator.hh

```diff
@@ -223,6 +220,32 @@ public:
      *  if receive buffers are known from previous run and have not changed
      *  communication could be faster */
     virtual void exchangeCached ( DataHandleInterface& ) const;
+
+  protected:
+    inline void computeDestinations( const linkage_t& linkage, vector_t& dest );
+
+    // return new tag number for the exchange messages
+    static int getMessageTag( const unsigned int increment )
+    {
+      static int tag = messagetag + 2 ;
+      // increase tag counter
+      const int retTag = tag;
+      tag += increment ;
+      // the MPI standard guaratees only up to 2^15-1
+      // this needs to be revised for the all-to-all communication
+      if( tag < 0 ) // >= 32767 )
```

Comment:
The only thing I concerned is that this thing is positive and somewhat unique. The rest I don't care. The int is only there, because that's what MPI wants.